common/random.h

```diff
@@ -60,6 +60,13 @@ class RandomSource {
 	uint32 getSeed() const { /*!< Get a random seed that can be used to initialize the RNG. */
 		return _randSeed;
 	}
+	
+	/**
+	* Scrambles the seed in order to get a new result.
+	* Code is shared between getRandomNumber and getRandomBit,
+	* so it is split off for clarity.
+	*/
+	inline void scrambleSeed();
```

Comment:
I would suggest to move this to the end of the class declaration and make it private.

Since c++ requires an inline function to be defined in every translation unit in which it is used, but you define it only in `random.cpp`, it is not a good idea to open the door to the possibility to use the function elsewhere. Making it private will ensure that does not happen.

Reply:
Fixed in commit 17c0c6f, and random.cpp was adjusted with commit aa765e4